### ISSUE TITLE
inference changes from version < 3.0.5 to 3.0.5 and nightly

### ISSUE THREAD

**Issue by innat** [keras-team/keras#19238]
To keras-team,

I got different logits of my model with latest release. Keras version 3.0.0 to 3.0.4 seems ok in this regard but from 3.0.5 to nightly it gives different logits and it's wrong. Following are the whole steps.

- [env Keras V2]: I have keras model in `.h5` format and able to get inference scores [[ **X** ]]. The weight was ported from torch weight.
- [env Keras V2]: I saved keras model in `.weights.h5` format and converted the model to keras 3.
- [env Keras V3]: I loaded `.weights.h5` format in new keras 3 model and loading was ok.
- [env Keras V3]: With `tf` and `jax` backend, the inference didn't match with [[ **X** ]] but did match with `torch` backend. In this case, the keras version was 3.0.4. In this case, the inference scores of `tf` and `jax` were similar and wrong w.r.t. officical weight.
- [env Keras V3]: From keras version 3.0.5 and nightly, the inference with `torch` doesn't match anymore with the original scores [[ **X** ]]. And now, `tf`, `jax`, and `torch` all give same scores and all of them are wrong w.r.t. official weight.

I took a loot at the changelogs, https://github.com/keras-team/keras/compare/v3.0.4...v3.0.5 but nothing suspicious. Any pointer?

**Comment by SuryanarayanaY** [comment]
Hi @innat ,
Thanks for reporting. If possible could you please submit a reproducible code snippet?

**Comment by innat** [comment]
@SuryanarayanaY This is bit tricky or somewhat complicated to reproduce this behaviour. Could u please take a look at [this comment](https://github.com/keras-team/keras-cv/pull/2369#issuecomment-1986143940)?

**Comment by innat** [comment]
Hello @SuryanarayanaY 
After debugging, I have found the root discrepancy of my issue. It is with the `tf.gather` and `keas.ops.take` methods. 

In keras 2

```python
data = np.ones((128, 5))
indices = [5, 5] 
gathered_values = tf.gather(data, indices=indices)
print(gathered_values, gathered_values.shape)
tf.Tensor(
[[1. 1. 1. 1. 1.]
 [1. 1. 1. 1. 1.]], shape=(2, 5), dtype=float64) (2, 5)
```

In keras 3.0.4 and lesser,

```python
data = np.ones((128, 5))
indices = [5, 5] 
gathered_values = ops.take(data, indices=indices)
print(gathered_values, gathered_values.shape)
tf.Tensor(
[[1. 1. 1. 1. 1.]
 [1. 1. 1. 1. 1.]], shape=(2, 5), dtype=float64) (2, 5)
```

In keras 3.0.5 and greater,

```python
data = np.ones((128, 5))
indices = [5, 5] 
gathered_values = ops.take(data, indices=indices)
print(gathered_values, gathered_values.shape)
tf.Tensor([1. 1.], shape=(2,), dtype=float64) (2,)
```

As you can see, output of keras==3.0.5 doesn't match with keras 2. To fix, we just needed to specify the axis explicitly, i.e. `ops.take(data, indices=indices, axis=0)`. 

(What a waste of time on this silly!)
